test/test_leaky_bucket_strategy.rb

```diff
@@ -39,6 +39,22 @@
       Prop::LeakyBucketStrategy.increment(@key, 5, interval: 1, threshold: 10)
       Prop::Limiter.cache.read(@key).must_equal bucket: 10, last_updated: @time.to_i
     end
+
+    it "adjusts bucket when elapsed time is less than interval" do
+      # bucket updated 5 seconds ago with leak rate of 1/second
+      Prop::Limiter.cache.write(@key, { bucket: 10, last_updated: @time.to_i - 5 })
+      Prop::LeakyBucketStrategy.increment(@key, 0, interval: 60, threshold: 60)
+      Prop::Limiter.cache.read(@key).must_equal bucket: 5, last_updated: @time.to_i
+    end
+
+    it "rounds down when applying leak amount to bucket" do
+      # this leaves the bucket at 5 rather than 6, leaning towards allowing more requests
+
+      # bucket updated 5 seconds ago with leak rate of 0.983/second
+      Prop::Limiter.cache.write(@key, { bucket: 10, last_updated: @time.to_i - 5 })
+      Prop::LeakyBucketStrategy.increment(@key, 0, interval: 60, threshold: 59)
+      Prop::Limiter.cache.read(@key).must_equal bucket: 5, last_updated: @time.to_i
+    end
```

Comment:
add a test here for the edge-case of decrementing by 0

Reply:
done